site/content/extend/plugins/webapp/actions.md

```diff
@@ -1,115 +1,493 @@
+**getTeamByName**
+
+Routes to the team of the specified `teamName` from the `Mattermost Server`.
 
-Now we can use `this.props.patchUser()` to update a user. The example component we made uses it to patch the current user's first name.
+```javascript
+  getTeamByName = (teamName: string) => {
+        this.trackEvent('api', 'api_teams_get_team_by_name');
+
+        return this.doFetch<Team>(
+            this.getTeamNameRoute(teamName),
+            {method: 'get'},
+        );
+    };
+```
+
+**executeCommand**
+
+Executes the specified command with the arguments provided and fetches the response.
+
+```javascript
+   executeCommand = (command: string, commandArgs: CommandArgs) => {
+        this.trackEvent('api', 'api_integrations_used');
+
+        return this.doFetch<CommandResponse>(
+            `${this.getCommandsRoute()}/execute`,
+            {method: 'post', body: JSON.stringify({command, ...commandArgs})},
+        );
+    };
+```
 
-To use our component in our plugin we would then use the registry in the initialization function of the plugin to register the component somewhere in the Mattermost UI. That is beyond the scope of this guide, but you can [read more about that here](/extend/plugins/webapp/reference/).
+**getOptions**
 
-## Available Actions
+Get the client options to make requests to the server. Use this to create your own custom requests.
+
+```javascript
+  getOptions(options: Options) {
+        const newOptions: Options = {...options};
+
+        const headers: {[x: string]: string} = {
+            [HEADER_REQUESTED_WITH]: 'XMLHttpRequest',
+            ...this.defaultHeaders,
+        };
+
+        if (this.token) {
+            headers[HEADER_AUTH] = `${HEADER_BEARER} ${this.token}`;
+        }
+
+        const csrfToken = this.csrf || this.getCSRFFromCookie();
+        if (options.method && options.method.toLowerCase() !== 'get' && csrfToken) {
+            headers[HEADER_X_CSRF_TOKEN] = csrfToken;
+        }
+
+        if (this.includeCookies) {
+            newOptions.credentials = 'include';
+        }
+
+        if (this.userAgent) {
+            headers[HEADER_USER_AGENT] = this.userAgent;
+        }
+
+        if (newOptions.headers) {
+            Object.assign(headers, newOptions.headers);
+        }
+
+        return {
+            ...newOptions,
+            headers,
+        };
+    }
+```
+
+## Custom reducers and actions
+
+Reducers in Redux are pure functions that describe how the data in the store changes after any given action. Reducers will always produce the same resulting state for a given state and action. You can register a custom reducer for your plugin against the Redux store with the `registerReducer` function.
+
+**registerReducer**
```

Comment:
Can we link to the [docs](https://developers.mattermost.com/extend/plugins/webapp/reference/#registerReducer) for `registerReducer` here?

Reply:
Link added for this one. Thanks.